pylint/message/message_definition_store.py

```diff
@@ -47,8 +47,14 @@ def register_message(self, message: MessageDefinition) -> None:
         self._messages_definitions[message.msgid] = message
         self._msgs_by_category[message.msgid[0]].append(message.msgid)
 
+    # We disable the message here because MessageDefinitionStore is only
+    # initialized once and due to the size of the class does not run the
+    # risk of creating a large memory leak.
+    # See discussion in: https://github.com/PyCQA/pylint/pull/5673
     @functools.lru_cache()
-    def get_message_definitions(self, msgid_or_symbol: str) -> List[MessageDefinition]:
+    def get_message_definitions(  # pylint: disable=lru-cache-decorating-method
```

Comment:
Do we want to raise on the function or on the decorator?

Reply:
I would say the decorator.

Reply:
Wow, this refactor was a lot of work, sorry about that !

Reply:
No problem! This is indeed much better.